### ISSUE TITLE
Raze not working - Redneck Rampage Collection - GOG

### ISSUE THREAD

**Issue by ChronoStriker1** [luxtorpeda-dev/packages#836]
### Game

Redneck Rampage Collection

### Engine

Raze

### Linux Distribution

SteamOS

### Luxtorpeda Version

v61

### Bug description

All testing done on steam deck running stable version of SteamOS.  I have a user-packages.json setup to use raze for the gog version of redneck rampage.  When attempting to use raze via steam nothing loads, when trying to then manually run the run-raze-rr.sh file it throws the error "./raze: error while loading shared libraries: libjpeg.so.62: cannot open shared object file: No such file or directory".  When placing the that library from another game into the lib folder the game will launch with the script.  

### Steps To Reproduce

1. Install GOG version of Redneck Rampage Collection through Lutris using GOG installer
2. Create link to Redneck Rampage folder named Rampage (did same to Redneck Rides Again folder named Again, but I don't think that's actually necessary.)
3. Start Redneck Rampage
4. Select Raze
5. Game will fail to load with file missing error for libjpeg.so.62

### Expected Behavior

Game works through steam.

### Additional Context

Note:  I have attempted to just add that file to an archive and host it myself but it will throw an error when trying to run through steam, but that will work.  It throws an error when attempting to run but if I manually run the startup script it will work.  Not sure where the issue is with that.

relevant portion of   user-packages.json
  
  "14109726720042991616": {
        "game_name": "Redneck Rampage",
        "download": [
            {
                "name": "rednukem",
                "url": "https://github.com/luxtorpeda-dev/packages/releases/download/rednukem-3/",
                "file": "rednukem-common-3.tar.xz",
                "cache_by_name": true
            },
            {
                "name": "raze",
                "url": "https://github.com/luxtorpeda-dev/packages/releases/download/raze-35/",
                "file": "raze-common-35.tar.xz",
                "cache_by_name": true
            }
        ],
        "choices": [
            {
                "name": "rednukem",
                "command": "./run-rr.sh",
                "download": [
                    "rednukem"
                ],
                "engine_name": "NBlood Rednukem"
            },
            {
                "name": "raze",
                "command": "./run-raze-rr.sh",
                "download": [
                    "raze"
                ]
            }
        ],
        "cloudNotAvailable": true
    }

### Relevant log output

```shell
When run through steam:

Raze 1.6.2 - 2023-01-07 19:45:11 +0100 - SDL version
Compiled on Jan  9 2023

Log started: 2023-01-30 11:56:37
OS: SteamOS, Linux 5.13.0-valve36-1-neptune on x86_64
 adding /home/deck/Games/redneck-rampage-collection/drive_c/GOG Games/Redneck Rampage Collection/raze.pk3, 2747 lumps
 adding /home/deck/Games/redneck-rampage-collection/drive_c/GOG Games/Redneck Rampage Collection/Redneck/REDNECK.GRP, 441 lumps
 adding /home/deck/Games/redneck-rampage-collection/drive_c/GOG Games/Redneck Rampage Collection/Redneck/, 166 lumps
CPU speed: 2800 MHz
CPU Vendor ID: AuthenticAMD
  Name: AMD Custom APU 0405
  Family 23 (23), Model 144, Stepping 2
  Features: SSE2 SSE3 SSSE3 SSE4.1 SSE4.2 AVX AVX2 F16C FMA3 BMI1 BMI2 HyperThreading
Loading "raze.pk3:engine/engine.def"

Loading "engine/widescreen.def"

I_InitSound: Initializing OpenAL
  Opened device Rembrandt Radeon High Definition Audio Controller Digital Stereo (HDMI 3)
  EFX enabled
No GENMIDI lump found. OPL playback not available.
script parsing took 83.43 ms
Compiling: 'engine/engine.con'.
Compiling: 'game.con'.
Compilation time:44.96 ms, Code Size:49156 bytes. 1325 labels. 0/512 Variables.
Using video driver x11
GL_VENDOR: AMD
GL_RENDERER: AMD Custom GPU 0405 (vangogh, LLVM 14.0.6, DRM 3.45, 5.13.0-valve36-1-neptune)
GL_VERSION: 4.6 (Core Profile) Mesa 22.2.0 (git-17e5312102) (Core profile)
GL_SHADING_LANGUAGE_VERSION: 4.60

Max. texture size: 16384
Max. texture units: 32
Max. varying: 128
Max. combined shader storage blocks: 80
Max. vertex shader storage blocks: 16
Resolution: 640 x 480
player 1 of 1 (1 nodes)
corrupted size vs. prev_size in fastbins
/home/deck/Games/redneck-rampage-collection/drive_c/GOG Games/Redneck Rampage Collection/run-raze-rr.sh: line 10: 41606 Aborted                 (core dumped) LD_LIBRARY_PATH="lib:$LD_LIBRARY_PATH" ./raze


Raze 1.6.2 - 2023-01-07 19:45:11 +0100 - SDL version
Compiled on Jan  9 2023

Log started: 2023-01-30 11:56:37
OS: SteamOS, Linux 5.13.0-valve36-1-neptune on x86_64
 adding /home/deck/Games/redneck-rampage-collection/drive_c/GOG Games/Redneck Rampage Collection/raze.pk3, 2747 lumps
 adding /home/deck/Games/redneck-rampage-collection/drive_c/GOG Games/Redneck Rampage Collection/Redneck/REDNECK.GRP, 441 lumps
 adding /home/deck/Games/redneck-rampage-collection/drive_c/GOG Games/Redneck Rampage Collection/Redneck/, 166 lumps
CPU speed: 2800 MHz
CPU Vendor ID: AuthenticAMD
  Name: AMD Custom APU 0405
  Family 23 (23), Model 144, Stepping 2
  Features: SSE2 SSE3 SSSE3 SSE4.1 SSE4.2 AVX AVX2 F16C FMA3 BMI1 BMI2 HyperThreading
Loading "raze.pk3:engine/engine.def"

Loading "engine/widescreen.def"

I_InitSound: Initializing OpenAL
  Opened device Rembrandt Radeon High Definition Audio Controller Digital Stereo (HDMI 3)
  EFX enabled
No GENMIDI lump found. OPL playback not available.
script parsing took 83.43 ms
Compiling: 'engine/engine.con'.
Compiling: 'game.con'.
Compilation time:44.96 ms, Code Size:49156 bytes. 1325 labels. 0/512 Variables.
Using video driver x11
GL_VENDOR: AMD
GL_RENDERER: AMD Custom GPU 0405 (vangogh, LLVM 14.0.6, DRM 3.45, 5.13.0-valve36-1-neptune)
GL_VERSION: 4.6 (Core Profile) Mesa 22.2.0 (git-17e5312102) (Core profile)
GL_SHADING_LANGUAGE_VERSION: 4.60

Max. texture size: 16384
Max. texture units: 32
Max. varying: 128
Max. combined shader storage blocks: 80
Max. vertex shader storage blocks: 16
Resolution: 640 x 480
player 1 of 1 (1 nodes)
corrupted size vs. prev_size in fastbins
/home/deck/Games/redneck-rampage-collection/drive_c/GOG Games/Redneck Rampage Collection/run-raze-rr.sh: line 10: 41606 Aborted                 (core dumped) LD_LIBRARY_PATH="lib:$LD_LIBRARY_PATH" ./raze


When manually run:

(deck@steamdeck Redneck Rampage Collection)$ ./run-raze-rr.sh 
./raze: error while loading shared libraries: libjpeg.so.62: cannot open shared object file: No such file or directoy
```

**Comment by d10sfan** [comment]
@ChronoStriker1 It seems like for some reason the libraries coming from the steam linux runtime are not being applied correctly on steam  when using a non-steam game. I don't own the game on GOG, so I don't really have a way to test this.

You may also want to try a different game and use the same process if you can, just to see if anything different happens.

Also, if you can go into the ~/.cache/luxtorpeda/appid folder and find the archive, edit the shell script to add an ldd line, to see what libraries it is detecting and where they are from, then paste that log output here. You'll want to update the archive, and then run the game through luxtorpeda again.

**Comment by ChronoStriker1** [comment]
> @ChronoStriker1 It seems like for some reason the libraries coming from the steam linux runtime are not being applied correctly on steam when using a non-steam game. I don't own the game on GOG, so I don't really have a way to test this.
> 
> You may also want to try a different game and use the same process if you can, just to see if anything different happens.
> 
> Also, if you can go into the ~/.cache/luxtorpeda/appid folder and find the archive, edit the shell script to add an ldd line, to see what libraries it is detecting and where they are from, then paste that log output here. You'll want to update the archive, and then run the game through luxtorpeda again.

I have tried blood with nblood previously and that worked as expected, doing the same with raze and I have the same symptoms as with redneck rampage.
As to you second test I have no appid folders in ~/.cache/luxtorpeda only in ~/.config/luxtorpeda I just want to make sure you didn't mean the name of the launcher app, there are those (that's how I was testing to add the libraries)

**Comment by d10sfan** [comment]
@ChronoStriker1 There should be some folders in there for the cache, the app id would be the one that you referenced in the user packages, can you take a screenshot of the folder listing of ~/.cache/luxtorpeda?

**Comment by ChronoStriker1** [comment]
Here is a screenshot
<img width="1401" alt="image" src="https://user-images.githubusercontent.com/2432904/216778504-2b518f18-9df1-4da7-85b0-3b334f9a270e.png">

Also it looks like the issue with the libraries not loading was actually a steam error, the issue was that the target in the steam properties was blank.  While working on trying lutorpeda I had changed from using lutris (since I used its gog install script) so that section had flatpak in it, I must have cleared it when trying out this as the loading method.  I changed that filed to point to the rampage.exe and it loads correctly for both raze and rednukem.  I only relized it when testing with blood as I actually had multiple steam entries and one was set with that field and the other I had cleared.
